app/screens/permalink/permalink.tsx

```diff
@@ -0,0 +1,222 @@
+    channel: ChannelModel;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        flex: 1,
+        marginTop: 20,
+    },
+    wrapper: {
+        backgroundColor: theme.centerChannelBg,
+        borderRadius: 6,
+        flex: 1,
+        margin: 10,
+        opacity: 1,
+    },
+    header: {
+        alignItems: 'center',
+        borderTopLeftRadius: 6,
+        borderTopRightRadius: 6,
+        flexDirection: 'row',
+        height: 44,
+        paddingRight: 16,
+        width: '100%',
+    },
+    dividerContainer: {
+        backgroundColor: theme.centerChannelBg,
+    },
+    divider: {
+        backgroundColor: changeOpacity(theme.centerChannelColor, 0.2),
+        height: 1,
+    },
+    close: {
+        justifyContent: 'center',
+        height: 44,
+        width: 40,
+        paddingLeft: 7,
+    },
+    titleContainer: {
+        alignItems: 'center',
+        flex: 1,
+        paddingRight: 40,
+    },
+    title: {
+        color: theme.centerChannelColor,
+        fontSize: 17,
+        fontWeight: '600',
+    },
+    postList: {
+        flex: 1,
+    },
+    loading: {
+        flex: 1,
+        justifyContent: 'center',
+        alignItems: 'center',
+    },
+    bottom: {
+        borderBottomLeftRadius: 6,
+        borderBottomRightRadius: 6,
+    },
+    footer: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        backgroundColor: theme.buttonBg,
+        flexDirection: 'row',
+        height: 43,
+        paddingRight: 16,
+        width: '100%',
+    },
+    jump: {
+        color: theme.buttonColor,
+        fontSize: 15,
+        fontWeight: '600',
+        textAlignVertical: 'center',
+    },
+    errorContainer: {
+        alignItems: 'center',
+        justifyContent: 'center',
+        padding: 15,
+    },
+    errorText: {
+        color: changeOpacity(theme.centerChannelColor, 0.4),
+        fontSize: 15,
+    },
+    archiveIcon: {
+        color: theme.centerChannelColor,
+        fontSize: 16,
+        paddingRight: 20,
+    },
+}));
+
+function Permalink({channel, postId, currentUsername}: Props) {
+    const [posts, setPosts] = useState<PostModel[]>([]);
+    const [loading, setLoading] = useState(true);
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+
+    const style = getStyleSheet(theme);
+
+    useEffect(() => {
+        (async () => {
+            const data = await fetchPostsAround(serverUrl, channel.id, postId);
+            if (data?.posts) {
+                setLoading(false);
+                setPosts(data.posts);
+            }
+        })();
+    }, []);
+
+    const handleClose = () => {
+        dismissModal();
+        closePermalink();
+    };
+
+    useEffect(() => {
+        const listener = BackHandler.addEventListener('hardwareBackPress', () => {
+            handleClose();
+            return true;
+        });
+
+        return () => {
+            listener.remove();
+        };
+    });
```

Comment:
Is it creating and attaching the listener on each render?

```suggestion
    }, []);
```